homework/index.js

```diff
@@ -29,22 +29,97 @@
     return elem;
   }
 
+  function createTableRows(table, optionsHeader = {}, optionsValue = {}) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('td', tr, optionsHeader);
+    if (optionsHeader.text === 'Repository:') {
+      const td = createAndAppend('td', tr);
+      createAndAppend('a', td, optionsValue);
+    } else {
+      createAndAppend('td', tr, optionsValue);
+    }
+  }
```

Comment:
Excellent that you saw an opportunity to make your code DRY by extraction the code repetition into a function that you can call repeatedly. This is a good first version, that I could imagine having written myself. But then, pondering this code the next day, I would probably set about doing another round of refactoring.

The idea of writing a function like this is to maximise the work that you can do inside the function and minimise the work that the caller still needs to do. For instance, I observe that all header text gets assigned the CSS class `td-header`. So you could hard-code that inside the function body and simply pass the text value for the header rather than an object:

```js
function createTableRows(table, header, optionsValue = {}) {
  const tr = createAndAppend('tr', table);
  createAndAppend('td', tr, { text: header, class: 'td-header' });
  if (header === 'Repository:') {
    const td = createAndAppend('td', tr);
    createAndAppend('a', td, optionsValue);
  } else {
    createAndAppend('td', tr, optionsValue);
  }
}
```

Then, because you need to do some special for the first row, you test whether the header `Repository:` is passed as the header text. This looks a bit awkward to me. It would probably take me another day to think of a more elegant solution. So the next day we could do a further refactoring. We can simplify the function to this (note the `return` statement): 

```js
function createTableRows(table, header, value) {
  const tr = createAndAppend('tr', table);
  createAndAppend('td', tr, { text: header, class: 'td-header' });
  createAndAppend('td', tr, { text: value });
  return tr;
}
```

But how then would we deal with that special first row? Answer: we can create the header and an empty value and stuff in the `<a>` tag afterwards:

```js
const firstRow = createTableRows(table, 'Repository:', '');
createAndAppend('a', firstRow.lastChild, {
  text: repo.name,
  href: repo.html_url,
  target: '_blank',
});
createTableRows(table, 'Description:', description);
createTableRows(table, 'Forks:', repo.fork);
createTableRows(table, 'Updated', formatDate(repo.updated_at));
```

Finally, since this function creates a single row (although you call it more than once), I would rename this function to `createTableRow` (drop the plural 's').

When I did this homework myself it took me several iterations over a couple of days to land at this final version.

Reply:
Well, my first version was also like this `createTableRows(table, header, options={});` but in order to add a tag I had to change it, which I see now was not very efficient at all. I guess I should have spend more time on it before sending the homework. It's a very nice solution to append `a` to `firstRow.lastChild`. I improved and implemented my code as you suggested. Thank you very much for the awesome feedback.

Reply:
Your solution was acceptable to me. But sometimes taking a little more time, if available, will bring new insights. Or somebody else looking at your code may trigger that. Don't worry if it's not 'perfect' this first time around.